tools/nl/eval/ndcg.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This module implements an
+# https://en.wikipedia.org/wiki/Discounted_cumulative_gain metric to score
+# ranked stat vars compared with a golden list.
+
+import numpy as np
+
+
+def calculate_relevance_scores(baseline_list):
+  num_tokens = len(baseline_list)
+  return {token: num_tokens - i for i, token in enumerate(baseline_list)}
+
+
+def dcg(scores, k=None):
```

Comment:
curious why k is needed here because it looks like it will always be None?

Reply:
+1

Reply:
Should try to use different k, meaning not comparing all elements. Need to get idea on what to use there.